src/mscorlib/src/System/Array.cs

```diff
@@ -1011,6 +1011,20 @@ public Object Clone()
             return ArraySortHelper<T>.Default.BinarySearch(array, index, length, value, comparer);
         }
 
+        public static T[] Clone<T>(T[] array)
+        {
+            if (array == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            }
+
+            // Note: We should always return a new instance, even for empty arrays.
```

Comment:
> We're concerned about Clone being used when lack of reference equality is important? Consistency with the instance Clone method?

Yep, pretty much. There have been lots of places recently where we've been optimizing to return empty arrays, e.g. in Linq or `"".ToCharArray()`. I left this note to make sure that optimization is not applied here.